app/uk/gov/hmrc/selfservicetimetopay/config/frontendWiring.scala

```diff
@@ -105,20 +110,22 @@ trait TimeToPayController extends FrontendController with Actions with CheckSess
   protected lazy val sessionCache: KeystoreConnector = SessionCacheConnector
   protected lazy val Action: ActionBuilder[Request] = checkSessionAction andThen PlayAction
   protected type AsyncPlayUserRequest = AuthContext => Request[AnyContent] => Future[Result]
-  protected lazy val authenticationProvider:GovernmentGateway = SaGovernmentGateway
+  protected lazy val authenticationProvider: GovernmentGateway = SaGovernmentGateway
   protected lazy val saRegime = SaRegime(authenticationProvider)
   private val timeToPayConfidenceLevel = new IdentityConfidencePredicate(ConfidenceLevel.L200, Future.successful(Redirect(routes.SelfServiceTimeToPayController.getUnavailable())))
   protected lazy val campaignManagerConnector: CampaignManagerConnector = CampaignManagerConnector
 
   def AuthorisedSaUser(body: AsyncPlayUserRequest): PlayAction[AnyContent] = AuthorisedFor(saRegime, timeToPayConfidenceLevel).async(body)
 
-  def authorizedForSsttp(block: => Future[Result])(implicit authContext: AuthContext, hc: HeaderCarrier): Future[Result] = {
-    campaignManagerConnector.isAuthorisedWhitelist(authContext.principal.accounts.sa.get.utr.utr).flatMap { isAuthorised =>
-      if (isAuthorised) {
-        block
-      } else {
-        Future.successful(Redirect(routes.SelfServiceTimeToPayController.getUnavailable()))
+  def authorizedForSsttp(block: (Option[TTPSubmission] => Future[Result]))(implicit authContext: AuthContext, hc: HeaderCarrier): Future[Result] = {
```

Comment:
authorizedForSSTTP function updated to perform the cache get of TTP data.
This allows us to access the data to ascertain the eligibility status from the cached data upfront and redirect to the 'call us page' in failure conditions.